### PR TITLE
feat: Terminate Workflow action will still execute exitHandler tasks

### PR DESCRIPTION
**Pull request by rushtehrani**
**What this PR does**:
Using Argo stop command instead of terminate allows the Workflow to still execute exitHandler tasks.

**Which issue(s) this PR fixes**:

Fixes onepanelio/core#

**Special notes for your reviewer**:

**Checklist**

Please check if applies

- [ ] I have added/updated relevant unit tests
- [ ] I have added/updated relevant documentation

Required 

- [x] I accept to release these changes under the Apache 2.0 License

### CODE REVIEW

**Comment by rushtehrani** [comment]
This also includes changes in https://github.com/onepanelio/core/pull/691, and should be merged after. Also it seems like it is marking the Workflows as failed in the UI instead of terminated.

File: pkg/workflow_execution.go [hunk 1]
```diff
@@ -914,7 +914,7 @@ func (c *Client) FinishWorkflowExecutionStatisticViaExitHandler(namespace, name
 			"finished_at": time.Now().UTC(),
 			"phase":       phase,
 		}).
-		Where(sq.Eq{"name": name}).
+		Where(sq.Eq{"name": name}, sq.NotEq{"phase": "Terminated"}).
```
Comment (aleksandrmelnikov) [inline pkg/workflow_execution.go:917]: @rushtehrani This needs to be changed to
```
		Where(sq.Eq{"name": name}).Where(sq.NotEq{"phase": "Terminated"}).
```
The 2nd parameter is ignored, and the "AND" piece is not added.
That's why it's not working.

**Review by aleksandrmelnikov — CHANGES_REQUESTED** [review]
Also, codacy is reporting a possible issue? I'm not seeing what the issue is though.

File: pkg/workflow_execution.go (hunk 1 above)
Comment (rushtehrani) [reply pkg/workflow_execution.go:917]: @aleksandrmelnikov good catch regarding `AND`, changing it to:
```go
               Where(sq.And{
			sq.Eq{"name": name},
			sq.NotEq{"phase": "Terminated"},
		}).
```

Comment (rushtehrani) [reply pkg/workflow_execution.go]: @aleksandrmelnikov issue is resolved, you can test with this image: `onepanel/core:4fec204`

Comment (aleksandrmelnikov) [reply pkg/workflow_execution.go]: Great, testing.

**Review by aleksandrmelnikov — APPROVED** [review]

**Pull request merged**